docs/en/building/supplemental.md

```diff
@@ -0,0 +1,541 @@
+				w(Outlet, { key: 'profile', id: 'profile', renderer: () => w(Profile, { username: 'Dojo User' }) })
+			])
+		]);
+	}
+}
+```
+
+The output will result in a separate bundle for each of the application's top level routes. In this example, there will be a main application bundle and bundles for `src/Home`, `src/About`, and `src/Profile`.
+
+To see automatic bundling in action create a new application using [@dojo/cli-create-app](https://github.com/dojo/cli-create-app/) and run `npm run build`. Dojo will automatically create bundles along the various routes in the sample application.
+
+## Manually specifying bundles
+
+Bundles may be manually specified in the `.dojorc` configuration file. This can be useful for breaking down an application into smaller bundles when automatic route bundling isn't sufficient.
+
+The `bundles` feature is part of the build app command. The configuration is comprised of a map of bundle names followed by a list of files or globs to match.
+
+For example, this configuration will bundle `About` and `Profile` together in a bundle named `additional.[hash].js`. Widget modules defined used with `w()` will be automatically converted to a lazily imported, local registry item in the parent widget. This provides a mechanism for declarative code splitting in your application.
+
+> .dojorc
+
+```json
+{
+	"build-app": {
+		"bundles": {
+			"additional": ["src/widgets/About", "src/widgets/Profile"]
+		}
+	}
+}
+```
+
+If we wanted to create nls internationalization modules by locale we could use globs to ensure all files under each language directory are included.
+
+> .dojorc
+
+```json
+{
+	"build-app": {
+		"bundles": {
+			"fr": ["src/**/nls/fr/**"],
+			"de": ["src/**/nls/de/**"]
+		}
+	}
+}
+```
+
+In this case Dojo will create bundles named `fr.[hash].js` and `de.[hash].js`. For more information see [Working with message bundles](/reference-guides/i18n/working-with-message-bundles) from the Internationalization reference guide.
+
+## Bundling considerations
+
+<!-- TODO I am not confident in what I am saying here. Under what conditions will duplication of common/shared resources occur? How do we avoid this? Can we define a bundle for common widgets? Should I talk about the [bundle analyzer](https://github.com/dojo/cli-build-app/blob/master/README.md#bundle-analyzer) -->
+
+Under the covers Dojo uses Webpack with a number of custom plugins to provide intelligent application bundling. However, sometimes decisions made by the build tool or manually defined in `.dojorc` can create duplication of common resources shared by multiple bundles. Some of this is unavoidable. A good general rule of thumb for avoid duplication is to try to ensure that common code is at the outermost edges of your dependency tree. In other words, minimize dependencies as much as possible among shared code. If a significant amount of code may be shared among bundles (e.g. common widgets) consider bundling these assets together.
```

Comment:
We shouldn't need to mention `webpack` at all, it's an implementation detail 😄

Reply:
This was included to address this bullet

> how cross-bundle duplication may occur for shared resources across multiple bundles

Should we just remove the reference to webpack and call it good, or should we say more/other things? Just to repeat my TODO above, I am not very confident in this section.